### PR TITLE
Get messages from availability channel

### PR DESCRIPTION
**Pull request by BlckHawker**
[Ticket](https://github.com/orgs/RIT-Crowd-Comic/projects/2?pane=issue&itemId=65030179)

### Description

Sets the availability channel. This will store the channel object in `utils/availability.js`.

#### Parameters
`channel` - The new availability channel. Only allows text channel

### Notes
N/A

### Screenshot(s)
image

### Known Issues / Warnings
This needs to be called at least once every time the bot is restarted. This won't be an issue once the DB is set up.
### Possible Additional Features
N/A

### Description / Notes
This event is called every time a message is sent in a channel the bot is in. (I assume this applies in all servers, but I have not tested this). If the `/set-availability-channel` has been ran correctly, all future messages in the availability channel will be parroted (excluding bots). This will be overwritten with the Open AI integration in a future PR

### Screenshot(s)
image

### Known Issues / Warnings
The try catch currently logs the error in the console. I would like to have it so the bot prints it out in the channel the command was sent, but that information is not easily avaible in this method
### Possible Additional Features
N/A

### CODE REVIEW

File: src/utils/apiCalls.js [hunk 1]
```diff
@@ -42,6 +42,19 @@ const getMessagesAfterId = async (channelId, limit, afterId, addFirstMessage = f
     return messages;
 }
 
+//returns an array of servers this bot is in
+const getServers = async () => {
+    return await getAPICall(`${baseUrl}/users/@me/guilds`)
+}
```
Comment (tstodter) [inline src/utils/apiCalls.js:48]: `await` is unnecessary here (as is `async`, technically).
```suggestion
const getServers = () => {
    return getAPICall(`${baseUrl}/users/@me/guilds`);
}
```

File: src/utils/availability.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+let availabilityChannel = undefined;
+const getAvailabilityChannel = () => {return availabilityChannel}
+const setAvailabilityChannel = (channel) => {availabilityChannel = channel}
+module.exports = {
+    getAvailabilityChannel,
+    setAvailabilityChannel
+}
```
Comment (tstodter) [inline src/utils/availability.js:8]: Let's add some attractive newlines in this file.

File: src/utils/availability.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+let availabilityChannel = undefined;
+const getAvailabilityChannel = () => {return availabilityChannel}
```
Comment (tstodter) [inline src/utils/availability.js:3]: Suggest we make this an async function. It doesn't need it right now, but I would anticipate that in the near future we'll need to make a db call to get this information, which will be async. So make it async now (just throw the async keyword on the front) for less refactoring later.

File: src/commands/availability/setAvailabilityChannel.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            channel_types: [ChannelType.GuildText]
+        }
+    ],
+
+    //logic
+    callback: async (_, interaction) => {
+        try {
+            await interaction.deferReply({ ephemeral: false })
+
+            const oldChannel = getAvailabilityChannel();
```
Comment (tstodter) [inline src/commands/availability/setAvailabilityChannel.js:22]: If we make it async, then `const oldChannel = await getAvailabilityChannel();`.

**Review by tstodter — APPROVED** [review]
LGTM, untested by I'm comfortable with a merge, after you adjust it to the new CommandBuilder style.

**Review by wct9667 — COMMENTED** [review]
I think you should refactor to the new system, although it won't cause merge issues. The files will not register with the new system and may throw errors. Also, in terms of the merge conflict with `apiCalls` keep your version, the changes on mine are from linting and will get fixed on commit anyway.

**Pull request merged**